File: android/src/main/java/app/independo/capacitorvoicerecorder/core/ResponseFormat.java
package app.independo.capacitorvoicerecorder.core;

import com.getcapacitor.PluginConfig;

/** Supported response payload shapes. */
public enum ResponseFormat {
    LEGACY,
    NORMALIZED;

    /** Reads the response format from plugin configuration. */
    public static ResponseFormat fromConfig(PluginConfig config) {
        String value = config.getString("responseFormat", "legacy");
        if ("normalized".equalsIgnoreCase(value)) {
            return NORMALIZED;
        }
        return LEGACY;
    }
}

File: ios/Sources/VoiceRecorder/Core/ResponseFormat.swift
import Foundation
import Capacitor

/// Supported response payload shapes.
enum ResponseFormat: String {
    case legacy
    case normalized

    /// Converts a raw config value into a response format.
    static func from(value: String?) -> ResponseFormat {
        guard let value = value?.lowercased(), value == "normalized" else {
            return .legacy
        }
        return .normalized
    }

    /// Reads the response format from plugin configuration.
    init(config: PluginConfig) {
        let value = config.getString("responseFormat", "legacy") ?? "legacy"
        self = ResponseFormat.from(value: value)
    }
}
